staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/custom.go

```diff
@@ -0,0 +1,342 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modes
+
+import (
+	"encoding"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+// Returns a non-nil error if and only if the argument's type (or one of its component types, for
```

Comment:
By component type doe we mean type embedding? (And we have tests for this?)

Reply:
I wasn't sure on canonical terminology for this, here I mean any type that is referenced in the declaration of a composite type. So the element type of a slice or array type, the pointed-to type of a pointer type, the types of struct fields, etc.

Reply:
OK, I think you have the right term for that.  I was thinking specifically about:

```
type X struct {} // implements one of the marshal/unmarshal interfaces
type Y struct { 
  X
}
```

And wanted to ensure we test for that case.

Reply:
Got it, I'll make sure that is covered.